keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -106,6 +124,58 @@ def pretokenize(text, lowercase, strip_accents, split):
     return text
 
 
+def download_vocabulary(
+    lang,
+    vocabulary_size,
+    lowercase,
+    strip_accents,
+    suffix_indicator,
+):
+    if lang not in SUPPORTED_VOCAB:
+        raise ValueError(
+            "This language is currently not supported. Recieved "
+            f"`lang={lang}`."
+        )
+    if strip_accents:
+        raise ValueError(
+            "The pretrained vocabularies currently do not support "
+            "`strip_accents=True`."
+        )
+    if suffix_indicator != "##":
+        raise ValueError(
+            "This suffix indicator is currently not supported in pretrained "
+            'vocabularies. Use the default `suffix_indicator="##"` or '
+            "provide your own vocabulary. Recieved "
+            f"`suffix_indicator={suffix_indicator}`."
+        )
+
+    # Get vocabulary file
+    if vocabulary_size is not None:
+        # 0.95 is from the 5% buffer when training vocabularies.
```

Comment:
Add a todo: shall we make a separate class PretrainedWordPieceTokenizer? These args seems to be messy, as we don't expect users to set those.

Reply:
I'd have preference for keeping the same class, as written here. This will keep it a similar UX to keras applications, where there is a single arg (weights in that case), that gets you a "pretrained" version of the object you are instantiating.

Adding a separate pretrained class would feel like a different UX.

Reply:
My concern is looking into the code, we have lots of code doing arg checking, and these log is not quite explicit. Assuming users to read thoroughly through the docstring may not actually work. 

One alternative I am thiking is we can make a new namespace `tokenizers.pretrained.WordPieceTokenizer`,  which only takes an identifer (maybe just `lang`).

Reply:
That's a good idea idea. If we were to follow the pattern of our new `Bert` code it would be `tokenizers.WordPieceTokenizerPretrained` which would wrap `tokenizers.WordPieceTokenizer`

I think `lang` is begging to be an `enum` here IMHO. Then we'd need almost no input validation.